astropy/table/table.py

```diff
@@ -1069,7 +1069,11 @@ def __array__(self, dtype=None):
         Coercion to a different dtype via np.array(table, dtype) is not
         supported and will raise a ValueError.
         """
-        if dtype is not None:
+        if np.dtype(dtype).kind == 'O':
+            out = np.array(None, dtype=object)
+            out[()] = self
+            return out
+        elif dtype is not None:
```

Comment:
Aside: I just removed this line and checked the example below on recent numpy and it now works as expected! (Separate PR, of course!)

Reply:
That'd be nice!